runtime/flangrti/llcrit.c

```diff
@@ -9,7 +9,13 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <string.h>
+#if !defined(_WIN32)
```

Comment:
pthread is indeed not in windows when you install it, but it doesn't mean there is no 3rd party implementations available, I've been using some of them in my past jobs to build some of the open source projects, with success.
Nevertheless, if Windows port can live without this header, then fine, but then, do we still need to refer 32-bit windows in our ifdefs? AFAIR there are new preprocessor definitions to indicate we're building for Windows...

Reply:
Yes, you're right, pthread 'is not available' doesn't mean that there is no other third-party implementation ( https://sourceware.org/pthreads-win32 ). It only means Windows itself doesn't natively support pthreads. For example pthread_atfork is not implemented in pthread-win32 library,  and it do not correspond well to the Microsoft windows API. Maybe we can use 'CreateProcess' Windows function to way around this later.

Regarding to the predefined macro I haven't seen any new Microsoft specific predefined macros.  In the documentation _WIN32 is generally used for detecting: "Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined." 
See: https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-160#microsoft-specific-predefined-macros

What about that: 
1) We defer it to CMake and add some user defined macros based on how CMake detects the OS and Arch. 
2) We add something like this (build_config.h) to Flang
#if defined(_WIN32)
...
#define OS_WIN 1
...
#endif
I prefer to defer these things to the compiler, but it's a matter of taste. Shall I open an issue for the further discussion?

Reply:
As I mentioned in some other review, I'm not working with Windows for years, I don't know what current recommendations are, apparently, the 64-bit specific macros I knew about are in Clang on Windows, but not in MSVC. I'd rather see an opinion of the others (potentially, more eager Windows users) than start a new discussion. For this review, I'll follow the approval decision of the majority.